kclvm/api/src/service/service_impl.rs

```diff
@@ -472,20 +473,22 @@ impl KclvmServiceImpl {
     ///
     /// let serv = KclvmServiceImpl::default();
     /// let result = serv.rename(&RenameArgs {
+    ///     package_root: "./src/testdata/rename".to_string(),
     ///     symbol_path: "a".to_string(),
-    ///     file_paths: vec!["./src/testdata/rename/main.k".to_string()],
+    ///     file_paths: vec!["main.k".to_string()],
     ///     new_name: "a2".to_string(),
     /// }).unwrap();
     /// assert_eq!(result.changed_files.len(), 1);
     /// ```
     pub fn rename(&self, args: &RenameArgs) -> anyhow::Result<RenameResult> {
+        let pkg_root = args.package_root.clone();
         let symbol_path = args.symbol_path.clone();
         let file_paths = args.file_paths.clone();
         let new_name = args.new_name.clone();
-        Ok(RenameResult {
-            //todo: mock implementation
-            changed_files: file_paths,
-        })
+
+        let changed_files =
+            rename::rename_symbol_on_file(&pkg_root, &symbol_path, &file_paths, new_name)?;
+        Ok(RenameResult { changed_files })
```

Comment:
Just use `rename::rename_symbol_on_file(&pkg_root, &symbol_path, &file_paths, new_name)` instead of `Ok(RenameResult { changed_files })`

Reply:
fixed